src/libsyntax/parse/parser/stmt.rs

```diff
@@ -473,4 +492,61 @@ impl<'a> Parser<'a> {
             "this was erroneously allowed and will become a hard error in a future release"
         }).emit();
     }
+
+    fn recover_stmt_local(
+        &mut self,
+        span: Span,
+        attrs: Vec<Attribute>,
+        msg: &str,
+    ) -> PResult<'a, Option<Stmt>> {
```

Comment:
I feel all of these methods should live in `parse/parser/diagnostics.rs`.

Reply:
Only the ones added in this PR?
There are other methods that do similar things, that were added in the past.

Reply:
I had a talk with @Centril and it's fine either way.